security-jwt/src/main/java/io/micronaut/security/token/jwt/signature/ec/ECSignatureGenerator.java

```diff
@@ -44,6 +48,13 @@ public class ECSignatureGenerator extends ECSignature implements SignatureGenera
     public ECSignatureGenerator(ECSignatureGeneratorConfiguration config) {
```

Comment:
I think it would be better to add a method `default Optional<String> getKid() { return Optional.empty() }` to `ECSignatureGeneratorConfiguration` instead.